hpcc/scripts/delete_ephemeral_storage_accounts

```diff
@@ -0,0 +1,16 @@
+#!/bin/bash
```

Comment:
TLH: When one asked for external storage accounts to be created, the 4 storage accounts that are created, by default, in the aks resource group should not be created (I assumed these are ephemeral). But 2 are created. This script deletes them IF you have asked for external storage accounts to be created.

Reply:
TLH: By default the HPCC module creates 4 storage accounts in the aks resource group. I have assumed these are epemeral storage accounts (mainly because Godji created the 'storage' module to create external storage which I assumed replaces the 4 storage accounts created in the aks resource group by default). These 4 storage accounts are used by just HPCC.

Reply:
_A reference to the offending code would be helpful._

Can you point out where those are created, please?  I'd like to take a look at this problem, because it makes more sense to avoid creating them then delete them after the fact.  Perhaps a PR can be created for an upstream repo that makes them totally optional.

Reply:
https://github.com/hpccsystems-solutions-lab/tlh-opinionated-terraform-azurerm-hpcc/blob/add-ecl-code-security-misc/storage.tf

Reply:
A couple of things:

1) If I'm reading that correctly, those storage accounts depend on:

`external_storage_config_enabled = (var.external_storage_config != null) && (var.internal_storage_enabled == false) ? true : false `

Your `hpcc.tf` file does not set either of those variables when referencing the module.

2) The link you pasted references a specific branch of that repo but your code does not.  I know that another PR request was to remove branches and it may have been this one, but the 'real' solution is to make sure the production code is on the main/master branch of the remote repo and then not reference a branch in your code.  Anyway, there may be a difference between the remote's main and specific branches, when it comes to storage account creation, and that should be explored.

The bottom line is, please try to avoid creating unneeded resources at all, rather than allowing creation and deleting after the fact.  The latter wastes time and money, and is error prone (your resource-gathering code could, logically, find something unrelated and delete it).

Reply:
internal_storage_enabled is set on line 49 of hpcc.tf
external_storage_config is set on line 69 of hpcc.tf.

Reply:
I just updated my copy of that project and see that there is no main/master branch.  Furthermore, the storage.tf file within 'add-ecl-code-security-misc' branch you had cited has not been updated in quite some time.

So this is clearly a non-trivial fix.  Please spend some time investigating and, as noted above, "_...please try to avoid creating unneeded resources at all, rather than allowing creation and deleting after the fact. The latter wastes time and money, and is error prone (your resource-gathering code could, logically, find something unrelated and delete it)._"

Reply:
In locals.tf, the following code should (I thought) keep the default storages from being created:
`  admin_services_storage = local.external_storage_config_enabled ? merge(
    {
      for plane in var.external_storage_config :
      plane.category => {
        size = plane.size,
        type = plane.storage_type
      } if plane.category != "data"
  }) : var.admin_services_storage
`

Reply:
In locals.tf, there are several other places I see 'local.external_storage_config_enabled' being used.

Reply:
Your last two replies are simple statements. Are there conclusions I should draw from them?  Do you have questions?  Is this issue resolved?

Reply:
I'm not sure what you want here. Did you want me to add a 'main' branch to the repo 'tlh-opinionated-terraform-azurerm-hpcc'?

Reply:
Please continue to work on this issue; leaving unresolved.